### PR TITLE
Change maven-publishing/multiple-publications to maven-publishing/customize-identity snippet

### PR DESCRIPTION
**Pull request by rieske**
This snippet was only embedded to demonstrate identity customization. As a result, it is now simplified to contain only what's required for the userguide and all the cross-configuration is removed.

### CODE REVIEW

File: subprojects/maven/src/integTest/groovy/org/gradle/api/publish/maven/SamplesMavenPublishIntegrationTest.groovy [hunk 1]
```diff
@@ -109,38 +109,6 @@ class SamplesMavenPublishIntegrationTest extends AbstractSampleIntegrationTest {
         dsl << ['groovy', 'kotlin']
     }
 
-    @Unroll
-    @UsesSample("maven-publish/multiple-publications")
-    def "multiple publications with #dsl dsl"() {
```
Comment (jjohannes) [inline subprojects/maven/src/integTest/groovy/org/gradle/api/publish/maven/SamplesMavenPublishIntegrationTest.groovy:114]: :o: We could keep the test but only check the one customised publication, right? I think this is actually valuable to test that the documentation is documenting what it says (publishing a jar with the expected name)

File: subprojects/docs/src/snippets/groovy/compilationAvoidance/groovy/build.gradle [hunk 2]
```diff
@@ -1,8 +1,5 @@
 subprojects {
     apply plugin: 'groovy'
-    repositories {
-        jcenter()
-    }
```
Comment (jjohannes) [inline subprojects/docs/src/snippets/groovy/compilationAvoidance/groovy/build.gradle:5]: :x: Are all the changes to "compilationAvoidance" in this PR expected. I think that snippet is already fixed in another PR?

**Review by jjohannes — CHANGES_REQUESTED** [review]

File: subprojects/docs/src/snippets/groovy/compilationAvoidance/groovy/build.gradle (hunk 2 above)
Comment (rieske) [reply subprojects/docs/src/snippets/groovy/compilationAvoidance/groovy/build.gradle]: Slipped in accidentally. Fixed now.

File: subprojects/maven/src/integTest/groovy/org/gradle/api/publish/maven/SamplesMavenPublishIntegrationTest.groovy (hunk 1 above)
Comment (rieske) [reply subprojects/maven/src/integTest/groovy/org/gradle/api/publish/maven/SamplesMavenPublishIntegrationTest.groovy:114]: Good idea. I refocused the test instead now on this changed use-case.

**Review by jjohannes — APPROVED** [review]

**Pull request merged**